miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -478,6 +530,33 @@ private void setIdentity(DetailedSample descendant) throws IOException, MisoNami
     child.setParent(identity);
   }
 
+  /**
+   * Finds the identity of a sample by climbing the hierarchy
+   * 
+   * @param sample
+   * @return identity attached to sample
+   * @throws IOException
+   */
+  private SampleIdentity getIdentity(DetailedSample sample) throws IOException {
+    if (sample.getParent() == null) {
+      throw new IllegalArgumentException(ERR_MISSING_PARENT_ID);
+    }
+    DetailedSample child = sample;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "child". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)